packages/eas-cli/src/commands/release/publish.ts

```diff
@@ -0,0 +1,152 @@
+import { getConfig } from '@expo/config';
+import { Command, flags } from '@oclif/command';
+import chalk from 'chalk';
+
+import { PublishMutation } from '../../graphql/mutations/PublishMutation';
+import log from '../../log';
+import { ensureProjectExistsAsync } from '../../project/ensureProjectExists';
+import {
+  findProjectRootAsync,
+  getProjectAccountNameAsync,
+  getReleaseByNameAsync,
+} from '../../project/projectUtils';
+import { buildUpdateInfoGroup, collectAssets, uploadAssetsAsync } from '../../project/publish';
+import { promptAsync } from '../../prompts';
+import { getLastCommitMessageAsync } from '../../utils/git';
+
+export default class ReleasePublish extends Command {
+  static description = 'Publish an update group to a release.';
+
+  static flags = {
+    inputDir: flags.string({
+      description: 'location of the bundle',
+      required: false,
+    }),
+    releaseName: flags.string({
+      description: 'current name of the release.',
+    }),
+    runtimeVersion: flags.string({
+      description: 'runtime version of the updates',
+      required: false,
+    }),
+    publishMessage: flags.string({
+      description: 'Short message describing the updates.',
+      required: false,
+    }),
+    json: flags.boolean({
+      description: `return a json with the edited release's ID and name.`,
+      default: false,
+    }),
+  };
+
+  async run() {
+    let {
+      flags: { json: jsonFlag, releaseName, runtimeVersion, publishMessage, inputDir },
+    } = this.parse(ReleasePublish);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      throw new Error('Please run this command inside a project directory.');
+    }
+    const accountName = await getProjectAccountNameAsync(projectDir);
+    const {
+      exp: { slug },
+    } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    if (!inputDir) {
+      const validationMessage = 'input directory must be specified.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ inputDir } = await promptAsync({
+        type: 'text',
+        name: 'inputDir',
+        message: 'Please enter the name of the directory created by the bundler:',
+        initial: 'dist',
+        validate: value => (value ? true : validationMessage),
+      }));
```

Comment:
Instead of prompting this (which is not a very good experience because the prompted name is not saved anywhere), I think `"dist"` should be the default value for the `--input-dir` param. This way if you can just run `expo export` with the default flags, followed by `eas release:publish` with default flags.

If the directory doesn't exist we can print something like:
```
The input directory "dist" does not exist.
You'll need to run a command to build the JS bundle first (e.g. `expo export`).
You can use `--input-dir` to specify a different input directory.
```